### PR TITLE
[DPE-4729] Separate network.publish_host

### LINKED ISSUE #334: non-leader instances won't settle to active/idle state
## Steps to reproduce

1. Deploy 3 opensearch nodes, and relate self-signed-certificates on top of MAAS cloud.

## Expected behavior
opensearch nodes should settle and reach active/idle state after a while.

## Actual behavior

Leader instance will settle shortly, but non-leaders get "Requesting lock on operation: start" & "Waiting for OpenSearch to start...".
status stuck at "waiting" for hours.

## Versions

Operating system: 

Juju CLI: 3.5.1

Juju agent: 3.5.1

Charm revision: 2/edge rev 102.

LXD: 

## Log output

Juju debug log: 

the only error I can find
unit-opensearch-2: 19:27:35 ERROR unit.opensearch/2.juju-log [Errno 111] Connection refused
unit-opensearch-2: 19:27:35 ERROR unit.opensearch/2.juju-log [Errno 111] Connection refused
unit-opensearch-1: 19:27:36 ERROR unit.opensearch/1.juju-log node-lock-fallback:0: [Errno 111] Connection refused
unit-opensearch-0: 19:29:47 ERROR unit.opensearch/0.juju-log opensearch-peers:1: [Errno 111] Connection refused
unit-opensearch-0: 19:29:47 ERROR unit.opensearch/0.juju-log opensearch-peers:1: Cannot connect to the OpenSearch server...
unit-opensearch-2: 19:29:47 ERROR unit.opensearch/2.juju-log opensearch-peers:1: [Errno 111] Connection refused

## Additional context

[opensearch.debug.log.gz](https://github.com/user-attachments/files/15932756/opensearch.debug.log.gz)

### PR DESCRIPTION
**Pull request by phvalguima**
Separate option `network.host` from `network.publish_host`. OpenSearch uses a single IP for transport, vs. its HTTP server for port `9200`. The `network.publish_host` forces to the right IP on the peer relation.

Removes the need to retrieve hostnames from `cm_ips` addresses on `add_seed_hosts`.

Updates tls_certificates lib.

Closes #334

### CODE REVIEW

File: lib/charms/opensearch/v0/opensearch_config.py [hunk 1]
```diff
@@ -230,15 +226,28 @@ def update_host_if_needed(self) -> bool:
...
-            return True
-
-        return False
+        NetworkHost = namedtuple("NetworkHost", ["entry", "old", "new"])
+
+        result = False
+        for host in [
+            NetworkHost(
+                "network.host",
+                set(self.load_node().get("network.host", [])),
```
Comment (Mehdi-Bendriss) [inline lib/charms/opensearch/v0/opensearch_config.py:235]: `self.load_node` can be externalized in a variable outside of the loop

File: tests/unit/lib/test_opensearch_config.py [hunk 2]
```diff
@@ -172,11 +171,5 @@ def test_set_node_and_cleanup_if_bootstrapped(self, gethostbyaddr):
         opensearch_conf = self.yaml_conf_setter.load(self.opensearch_yml)
         self.assertNotIn("cluster.initial_cluster_manager_nodes", opensearch_conf)
 
-        # test unicast_hosts content
```
Comment (Mehdi-Bendriss) [inline tests/unit/lib/test_opensearch_config.py:175]: why was this test deleted?

**Review by Mehdi-Bendriss — COMMENTED** [review]
Thanks Pedro - I have a couple of comments

File: lib/charms/tls_certificates_interface/v3/tls_certificates.py
Comment (Mehdi-Bendriss) [inline lib/charms/tls_certificates_interface/v3/tls_certificates.py:1]: Let's not update this for now please, my next PR will actually downgrade it - due to an issue I've been witnessing which triggers endless secret changed events. Let's upgrade it in another PR once we confirm the issue

File: tests/unit/lib/test_opensearch_config.py (hunk 2 above)
Comment (Mehdi-Bendriss) [reply tests/unit/lib/test_opensearch_config.py:175]: Understood, i think we can keep it and just remove the hostnames from the `expected` collections, and exclusively keep the IP

**Review by reneradoi — DISMISSED** [review]
That is a nice integration test!

**Review by Mehdi-Bendriss — APPROVED** [review]

**Review by reneradoi — APPROVED** [review]

**Pull request merged**